### PR TITLE
CORE-414 Bad signature -> Mistrust

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/message_api.c [hunk 1]
```diff
@@ -1864,7 +1864,7 @@ static PEP_rating decrypt_rating(PEP_STATUS status)
     case PEP_DECRYPTED:
     case PEP_VERIFY_SIGNER_KEY_REVOKED:
     case PEP_DECRYPT_SIGNATURE_DOES_NOT_MATCH:
-        return PEP_rating_unreliable;
```
Comment (ip-plank-security) [inline src/message_api.c:1867]: seems like a trivial change. I briefly tried to look up PEP_DECRYPT_SIGNATURE_DOES_NOT_MATCH, and I think the status is set by sequoia, correct?

Comment (saschabacardit) [reply src/message_api.c:1867]: Shouldn't we move this to

```
case PEP_DECRYPTED:
case PEP_VERIFY_SIGNER_KEY_REVOKED:
    return PEP_rating_unreliable;
case PEP_DECRYPT_SIGNATURE_DOES_NOT_MATCH:
    return PEP_rating_mistrust;
```

Comment (ghost) [reply src/message_api.c:1867]: This is the whole trail for PEP_DECRYPT_SIGNATURE_DOES_NOT_MATCH (yes, set by sequoia):

https://github.com/plancksecurity/foundation-planckCoreSequoiaBackend/blob/v3.3.28/src/lib.rs#L1007
https://github.com/plancksecurity/foundation-planckCoreSequoiaBackend/blob/v3.3.28/src/pep.rs#L127
https://github.com/plancksecurity/foundation-planckCoreV3/blob/v3.3.28/src/pEpEngine.h#L156

Comment (ghost) [reply src/message_api.c:1867]: The tickets talks about "invalid signatures". No signature (PEP_DECRYPTED) and one from a revoked key (PEP_VERIFY_SIGNER_KEY_REVOKED) are counted as "invalid". Please note that this is only for encrypted mails.

**Review by ip-plank-security — APPROVED** [review]

**Review by saschabacardit — APPROVED** [review]

**Pull request merged**